src/objects/zcl_abapgit_objects_program.clas.abap

```diff
@@ -722,6 +852,74 @@ CLASS zcl_abapgit_objects_program IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD serialize_all_docu.
+    DATA: lt_docu_cat TYPE ty_oo_docu_cat_tt. " catalog of component docu
+
+    DATA lv_pattern TYPE string.
+    DATA lt_langu_additional TYPE zif_abapgit_lang_definitions=>ty_langus.
+    DATA lr_id TYPE RANGE OF dokhl-id.
+    DATA ls_id LIKE LINE OF lr_id.
+    DATA: BEGIN OF ls_dokhl,
+            id     TYPE doku_id,
+            object TYPE doku_obj,
+          END OF ls_dokhl.
+    DATA ls_docu_cat TYPE ty_oo_docu_cat.
+
+    ls_id-sign = 'I'.
+    ls_id-option = 'EQ'.
+    ls_id-low = iv_id.
+    APPEND ls_id TO lr_id.
+
+    IF iv_id = 'CL'.
+      ls_id-low = 'CA'.
+      APPEND ls_id TO lr_id.
+      ls_id-low = 'CE'.
+      APPEND ls_id TO lr_id.
+      ls_id-low = 'CO'.
+      APPEND ls_id TO lr_id.
+    ELSE.
+      ls_id-low = 'IA'.
+      APPEND ls_id TO lr_id.
+      ls_id-low = 'IE'.
+      APPEND ls_id TO lr_id.
+      ls_id-low = 'IO'.
+      APPEND ls_id TO lr_id.
+    ENDIF.
+
+    CONCATENATE iv_clsname '%' INTO lv_pattern RESPECTING BLANKS.
```

Comment:
use `%` only for subcomponents and `%` should be on pos 30. otherwise `cl_test%` will match `cl_test_2` as well.